`TestResolveDefaultAddrHTTPProxy` flakiness

https://github.com/gravitational/teleport/actions/runs/5213484920

```
=== RUN   TestResolveDefaultAddrHTTPProxy
2023-06-08T16:33:37Z DEBU [TSH]       Resolving default proxy port (insecure: true) common/resolve_default_addr.go:111
2023-06-08T16:33:37Z DEBU [TSH]       Trying 172.18.0.2:41619... common/resolve_default_addr.go:99
2023/06/08 16:33:37 http: TLS handshake error from 172.18.0.2:44890: tls: first record does not look like a TLS handshake
make: *** [Makefile:735: test-go-tsh] Error 1
2023-06-08T16:33:37Z DEBU [TSH]       Proxy address test failed error:[Get "https://172.18.0.2:41619/webapi/ping": EOF] common/resolve_default_addr.go:63
2023-06-08T16:33:37Z DEBU [TSH]       Waiting for all in-flight proxy address tests to finish common/resolve_default_addr.go:139
    resolve_default_addr_test.go:299: 
        	Error Trace:	/__w/teleport/teleport/tool/tsh/common/resolve_default_addr_test.go:299
        	Error:      	Received unexpected error:
        	            	Get "https://172.18.0.2:41619/webapi/ping": EOF
        	Test:       	TestResolveDefaultAddrHTTPProxy
--- FAIL: TestResolveDefaultAddrHTTPProxy (0.01s)
```

--- a/api/testhelpers/proxy.go
+++ b/api/testhelpers/proxy.go
@@ -21,7 +21,6 @@ import (
 	"net/http/httptest"
 	"sync"
 	"testing"
-	"time"
 
 	"github.com/gravitational/trace"
 	"github.com/stretchr/testify/require"
@@ -88,9 +87,9 @@ func (p *ProxyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	go replicate(sconn, dconn)
 	go replicate(dconn, io.MultiReader(buf, sconn))
 
-	// Wait until done, error, or 10 second.
+	// Wait until done.
 	select {
-	case <-time.After(10 * time.Second):
+	case <-r.Context().Done():
 	case <-errc:
 	}
 }

